data-model-translator-wrapper/src/main/java/ai/aitia/arrowhead/dmtpw/service/DataModelTranslationService.java

```diff
@@ -0,0 +1,162 @@
+	// members
+	
+	@Autowired
+	private DataModelTranslatorWrapperSystemInfo sysInfo;
+
+	@Autowired
+	private DataModelTranslationServiceValidation validator;
+
+	private final BlockingQueue<DataModelTranslationTask> jobQueue = new LinkedBlockingQueue<>();
+	private final Map<UUID, DataModelTranslationTask> jobCache = new ConcurrentHashMap<>();
+	private final Map<UUID, String> fileNameCache = new ConcurrentHashMap<>();
+
+	private final Logger logger = LogManager.getLogger(this.getClass());
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+    public UUID initTranslation(final DataModelTranslationInitRequestDTO dto, final String origin) {
+    	logger.debug("initTranslationJob started...");
+    	
+    	final DataModelTranslationInitRequestDTO normalized = validator.validateAndNormalizeDataModelTranslationInitRequestDTO(dto, origin);
+
+        final DataModelTranslationTask job = new DataModelTranslationTask(sysInfo.getModelIdsWithResultMimeTpyes().get(List.of(normalized.inputModelId(), normalized.outputModelId())), normalized.payload());
+        jobQueue.add(job);
+        jobCache.put(job.getUuid(), job);
+        return job.getUuid();
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    public DataModelTranslationResultResponseDTO getTranslationResult(final UUID jobId, final String origin) {
+    	logger.debug("getTranslationResult started...");
+
+        final DataModelTranslationTask job = jobCache.get(jobId);
+        
+        if (job == null) {
+        	throw new InvalidParameterException("Invalid UUID", origin);
+        }
+
+        final String result = readTranslatedFile(fileNameCache.get(jobId));
+        return new DataModelTranslationResultResponseDTO(job.getStatus(), result, job.getResultMimeType());
+    }
+
+	//=================================================================================================
+	// assistant methods
+
+    //-------------------------------------------------------------------------------------------------
+    private void doTranslationJob(final DataModelTranslationTask job) {
+    	// start the job
+    	job.setStatus(DataModelTranslationTaskStatus.IN_PROGRESS);
+
+    	// decode the bytes
+    	final byte[] bytes = Base64.getDecoder().decode(job.getPayload());
+
+    	final String fileName = job.getUuid().toString() + ".xml";
+    	try {
+    		// save the file
+    		Files.write(Paths.get(sysInfo.getInputFolder(), fileName), bytes);
+
+    		// run the script
+    		runTranslationScript(job.getUuid().toString());
+    	} catch (final Exception ex) {
+    		job.setStatus(DataModelTranslationTaskStatus.ERROR);
+    		throw new InternalServerError("An error occured while running the translation script");
```

Comment:
log the original exception
set the error message in the job as well
maybe add ex.getMessage() to your exception message